### DISCUSSION TITLE
Selecting from many to many

### DISCUSSION THREAD

**Discussion by rodjoseph** [supabase/supabase#1685] [Questions] [answered]
I'm having issues with selecting from a many-to-many relationship.

I have a table called items and a table called collections, and I would like to select all of the items in a collection.
```const fetchCollectionItems = async () => {
    const {data, error} = await supabase
    .from('collection_items').select('items(*)').eq('collection', route.params.collection.id)
    setItems(data)
  }
  ```

The data return looks like this:
```
Array [
  Object {
    "items": Object {
      "created_at": "2021-05-22T22:28:21.872165+00:00",
      "created_by": "75dc9cdf-8a97-4768-b006-772629ed5c1d",
      "description": null,
      "for_sale": false,
      "id": "b397429f-69c5-46ea-b8a6-7acddaff9e20",
      "image_url": null,
      "name": "Cosmogramma",
      "type": "Music",
      "updated_at": "2021-05-22T22:28:21.872165+00:00",
    },
  },
  Object {
    "items": Object {
      "created_at": "2021-03-28T04:05:51.693682+00:00",
      "created_by": "75dc9cdf-8a97-4768-b006-772629ed5c1d",
      "description": "...",
      "for_sale": null,
      "id": "2cea75bd-249b-4355-9100-3b42816f41ea",
      "image_url": "https://media.pitchfork.com/photos/6011c71da93b02093530eb0b/1:1/w_320/Madlib.jpg",
      "name": "Sound Ancestors",
      "type": "Unspecified",
      "updated_at": "2021-05-22T15:25:38.067605+00:00",
    },
  },
]
```

What is returned is accurate, but I can't use `data["items"]` or `data.items` and return the inner list(s)? Ideally I would like one list with all the items in the collection

**Comment by soedirgo** [comment, marked as answer]
Hi @rodjoseph, you can alternatively do this to have all the items in a list:
```js
const { data, error } = await supabase
    .from('collections')
    .select('items(*)')
    .eq('id', route.params.collection.id)
setItems(data[0].items)
```

**Reply by rodjoseph** [reply]
Hi @soedirgo, thanks for your answer. Unfortunately, when I try to log `data` in that case, the result is null

**Reply by soedirgo** [reply]
Ah, my bad, it should index to the first entry of `data` first. Updated the snippet above. Might need a check for when `data` is empty, but you get the idea.

**Reply by rodjoseph** [reply]
I'm getting `[Unhandled promise rejection: TypeError: null is not an object (evaluating 'data[0]')]`

**Reply by soedirgo** [reply]
Can you check what the value of `error` is?

**Reply by rodjoseph** [reply]
It says `error: More than one relationship was found for collections and items`. Hmmmm... I will investigate this.

**Reply by rodjoseph** [reply]
Only place where I have both ids in the same row is in collection_items

**Reply by rodjoseph** [reply]
> 
> 
> Only place where I have both ids in the same row is in collection_items

Actually, there is another place where they are in the same row too. Is there a way to specify which table to use?

**Reply by soedirgo** [reply]
I see, in this case you need to disambiguate the relation ([ref](https://postgrest.org/en/stable/api.html#embedding-disambiguation)). Can you try this one?
```js
const { data, error } = await supabase
    .from('collections')
    .select('items!collection_items(*)')
    .eq('id', route.params.collection.id)
setItems(data[0].items)
```

**Reply by rodjoseph** [reply]
Thank you very much. It worked after a couple of tweaks. So much to learn about PostgREST

**Reply by soedirgo** [reply]
Glad I can help! :)
